.ci/xenial.jenkinsfile

```
@@ -0,0 +1,148 @@
+// These are things to parameterize/read/write from/to .env file
+List gccVersions = ['5', '6', '7', '8', '9', '10', '11']
+List clangVersions = ['10', '11', '12']
+boolean isMaster = (scm.branches[0].name == 'master')
+echo ">> scm.branches[0].name: ${scm.branches[0].name}"
+List<String> environment = []
+
+def configure_docker() {
+    withCredentials([usernamePassword(credentialsId: 'c3istg-dad-docker', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
```

```suggestion
    withCredentials([usernamePassword(credentialsId: 'c3istg-dad-docker', usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
```

What about the official hub.docker registry? People will download from there, at least, we could upload a copye. I know using our side could save CI time as all is at same server, but still don't want to configure a new registry only to download a single image.

---

Sure, PRs should be cached on our side, when the PR is merged to `master` then those images will be built again (or just promoted) to DockerHub.